pkg/logs/internal/decoder/auto_multiline_detection/user_samples.go

```diff
@@ -45,24 +46,32 @@ type UserSamples struct {
 }
 
 // NewUserSamples creates a new UserSamples instance.
-func NewUserSamples(config model.Reader) *UserSamples {
+func NewUserSamples(config model.Reader, sourceSamples []*config.AutoMultilineSample) *UserSamples {
 	tokenizer := NewTokenizer(0)
 	s := make([]*UserSample, 0)
 	var err error
-	raw := config.Get("logs_config.auto_multi_line_detection_custom_samples")
-	if raw != nil {
-		if str, ok := raw.(string); ok && str != "" {
+	rawMainSamples := config.Get("logs_config.auto_multi_line_detection_custom_samples")
+	if rawMainSamples != nil {
+		if str, ok := rawMainSamples.(string); ok && str != "" {
 			err = json.Unmarshal([]byte(str), &s)
 		} else {
 			err = structure.UnmarshalKey(config, "logs_config.auto_multi_line_detection_custom_samples", &s)
 		}
 
 		if err != nil {
-			log.Error("Failed to unmarshal custom samples: ", err)
+			log.Error("Failed to unmarshal main config custom samples: ", err)
 			s = make([]*UserSample, 0)
 		}
 	}
 
+	for _, sample := range sourceSamples {
```

Comment:
I think samples are applied in order. Do you think it makes sense to prepend the source samples such that they take precedence over the global samples? That way if a user has a global sample/regex, and then supplies a more specific sample/regex on the source, it will override the global one.

Reply:
Yeah that makes sense, changing it now.